Can not link igraph library

When trying to build the addon, the linking process fails:

```
make[3]: *** No rule to make target '/libigraph.so', needed by 'programs/libcirkit_igraph.so'. Stop.
```
The problem is that you try to link against 
`${PC_LIBIGRAPH_LIBRARY_DIRS}/libigraph${CMAKE_SHARED_LIBRARY_SUFFIX}`. The variable `IMPORTED_TARGET` is empty on my system:
```
$ pkg-config --libs igraph
-ligraph
$ pkg-config --libs-only-L igraph

$
```

The following `CMakeLists.txt` lets me compile and link the addon:
```
find_package(PkgConfig)
pkg_check_modules(PC_LIBIGRAPH REQUIRED IMPORTED_TARGET igraph)

add_cirkit_library(
  NAME cirkit_igraph
  AUTO_DIRS src
  USE
    cirkit_classical
    cirkit_cli
    PkgConfig::PC_LIBIGRAPH
  INCLUDE
    PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}/src
    PUBLIC ${PC_LIBIGRAPH_INCLUDE_DIRS}
  DEFINE
    PUBLIC ADDON_IGRAPH
  COMMANDS
    classical/cli/commands/igraph_commands.hpp
)
```
My change is to use the `IMPORTED_TARGET` option that defines `PkgConfig::PC_LIBIGRAPH` as a valid target for cmake's `target_link_libraries()`.

I don't know whether this a general issue or somehow bound to my system. 
I am running Xubuntu 17.04, installed `libigraph0-dev` and `libigraph0v5` and use cmake version 3.7.2 and pkg-config version 0.29.1.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,5 +1,5 @@
 find_package(PkgConfig)
-pkg_check_modules(PC_LIBIGRAPH QUIET igraph)
+pkg_check_modules(PC_LIBIGRAPH REQUIRED IMPORTED_TARGET igraph)
 
 add_cirkit_library(
   NAME cirkit_igraph
@@ -7,7 +7,7 @@ add_cirkit_library(
   USE
     cirkit_classical
     cirkit_cli
-    ${PC_LIBIGRAPH_LIBRARY_DIRS}/libigraph${CMAKE_SHARED_LIBRARY_SUFFIX}
+    PkgConfig::PC_LIBIGRAPH
   INCLUDE
     PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}/src
     PUBLIC ${PC_LIBIGRAPH_INCLUDE_DIRS}

--- a/src/classical/cli/commands/dswap.cpp
+++ b/src/classical/cli/commands/dswap.cpp
@@ -42,6 +42,9 @@ dswap_command:: dswap_command( const environment::ptr& env )
 {
   opts.add_options()
     ( "num_vars,n", value_with_default( &num_vars ), "number of variables" )
+          ("abstract","work on the equivalence classes")
+          ("hamiltonian", "check for and compute a hamiltonian path")
+          ("hamilenum", "enumerate hamiltonian paths")
     ( "frompath",   value( &frompath ),              "generate delta-swap sequence from path" )
     ( "dotname",    value( &dotname ),               "write graph to this file" )
     ;
@@ -61,6 +64,12 @@ bool dswap_command::execute()
   {
     settings->set( "frompath", frompath );
   }
+
+  settings->set("abstract", is_set("abstract"));
+  settings->set("hamiltonian", is_set("hamiltonian"));
+  settings->set("hamilenum", is_set("hamilenum"));
+
+
 
   enumerate_linear_transformations( num_vars, settings );
 

--- a/src/classical/enumeration/enumerate_linear_transformations.cpp
+++ b/src/classical/enumeration/enumerate_linear_transformations.cpp
@@ -347,7 +347,7 @@ unsigned compute_order( unsigned n )
 {
   unsigned order = 1u;
 
-  for ( auto i = 0; i < n; ++i )
+  for ( unsigned i = 0; i < n; ++i )
   {
     order *= ( 1 << n ) - ( 1 << i );
   }
@@ -646,7 +646,7 @@ void enumerate_linear_transformations( unsigned n, const properties::ptr& settin
     } );
 
   /* create edges */
-  for ( auto v = 0; v < node_to_matrix.size(); ++v )
+  for ( size_t v = 0; v < node_to_matrix.size(); ++v )
   {
     const auto& bm = node_to_matrix[v];
 
@@ -678,9 +678,9 @@ void enumerate_linear_transformations( unsigned n, const properties::ptr& settin
     /* swapping */
     if ( !abstract )
     {
-      for ( auto row2 = 1; row2 < n; ++row2 )
+      for ( unsigned row2 = 1u; row2 < n; ++row2 )
       {
-        for ( auto row1 = 0; row1 < row2; ++row1 )
+        for ( unsigned row1 = 0u; row1 < row2; ++row1 )
         {
           const auto other = bm.swap( row1, row2 );
           const auto w = matrix_to_node[other.value()];

